tests/infra_test.go

```diff
@@ -975,3 +958,28 @@ func getKeysFromMetrics(metrics map[string]float64) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func checkError(err error) error {
+	var e *neturl.Error
+	if errors.As(err, &e) && e.Timeout() {
+		return nil
+	} else {
+		return err
+	}
+}
+
+func waitHttpResponse(errorsChan chan error, concurrency int) (int, error) {
+	errorCount := 0
+	var sampledErr error
+	for ix := 0; ix < concurrency; ix++ {
+		err := <-errorsChan
```

Comment:
i might add a timer for this as well, in case the workers have panic, or we can trust them ?